datamade/just-spaces
# Allow the user to create bar charts in the DAD

## Overview

Closes #96, closes #97 

Add logic for displaying bar charts for surveys in the data analysis designer.

## Demo

![test](https://user-images.githubusercontent.com/14170650/57645670-cb686b00-758c-11e9-93ee-1725db3c930d.gif)

## Notes

There's officially enough data processing logic in the DAD (particularly in `charts.js`) that it would make sense for us to have tests. I didn't want to have to add the task of configuring a whole JavaScript test runner to the scope of this PR, so I opened up #131 instead.

## Testing Instructions

 * Make sure you have surveys with questions and responses of each question type:
    * A custom Number (e.g. "How many fingers do you have?")
    * A preset observational question (e.g. `gender_observational`)
    * A preset intercept question with dropdown responses (e.g. "What is your gender?")
    * A preset intercept question with free response (e.g. "How old are you?")
* On the `Collected data` view, try making charts for each question type and confirm that they look right
